server/src/middleware/editComment.middleware.ts

```diff
@@ -0,0 +1,30 @@
+import RequestWithUser from '../interfaces/requestWithUser.interface';
+import { NextFunction, Response } from 'express';
+import UserDoesNotHavePermissionToExecutedRequestedData from '../exceptions/UserDoesNotHavePermissionToExecutedRequestedData';
+import commentsModel from '../models/comments.model';
+import CommentIsNotExist from '../exceptions/CommentIsNotExist';
+import UserDoesNotHavePermissionToEditThisComment from '../exceptions/UserDoesNotPermissiontoEditThisComment';
+
+async function commentEditMiddleware(request: RequestWithUser, response: Response, next: NextFunction) {
+  const commentId = request.params.id;
+  let selectedComment = null;
+  const userRole = request.user.userRole;
+  const userId = request.user._id;
+  const comment = commentsModel;
+
+  try {
+    selectedComment = await comment.findById(commentId);
+    const commentOwner = selectedComment.user !== null && selectedComment.user;
+    if ( userRole === 2) {
```

Comment:
```suggestion
    if ( userRole === userRoles.User) {
```
We have to make a map that will hold these numeric values